home/tests/test_links_in_page/test_external_links.py

```diff
@@ -18,8 +23,25 @@ def create_published_article_to_root(self, body) -> Article:
         return article
 
     def test_if_rich_text_field_internal_links_redirect_directly(self):
-        article = self.create_published_article_to_root(body="some BODY")
+        body = [
+            ("paragraph", RichText('<p><a href="http://aynthing.com">external link on beginning</a> dolor sit amet</p>')),
+            ("paragraph", RichText('<p>Lorem ipsum <a href="http://aynthing.com">external link in middle</a> dolor sit amet</p>')),
+            ("paragraph", RichText('<p>Lorem dolor sit amet <a href="http://aynthing.com">external link at the end</a></p>')),
+        ]
+
+        article = self.create_published_article_to_root(body=body)
         response = self.client.get(article.get_url())
 
+        assert response.status_code == status.HTTP_200_OK
+
+        html_parser = BeautifulSoup(response.content, "html.parser")
```

Comment:
@mwrobel-merixstudio How do you feel about using `assertTempalteUsed` and assert-ing on `response.context`?

Here's a [https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Testing#views](link) on how that can be done.

Reply:
I don't think this is part of this test - we want to test only parsing content, and which template we are using template isn't relevant in this test / code implementation. this test should:
- prepare proper Article
- make sure number of links source of Article is same as number in rendered Article
- make sure only links we want to be parsed are parsed
- make sure we parse every link that lead outside Wagtail

if there is possibility to use different templates to render Article - yes, then we should add template assert. @mikowhy is this change of template from admin panel possible?

Reply:
Got it. Thanks